### PR TITLE
LinterTest.py: Use assertRaisesRegex

### PR DESCRIPTION
**Pull request by madhukar01**
Usage of assertRaises was replaced by assertRaisesRegex which has
an extra parameter for exception message

Related to [coala#3374](https://github.com/coala/coala/issues/3374)

### CODE REVIEW

**Comment by RohanVB** [comment]
Hey @Madhukar-SH I don't think you followed: http://api.coala.io/en/latest/Developers/Newcomers_Guide.html
Check it out, and ping on gitter if you need any help :)

**Comment by Madhukar-SH** [comment]
@RohanVB I have finshed that quite long back

**Comment by RohanVB** [comment]
Please follow the instructions given in it, it mentions that `coala` should be run on the project before pushing to get rid of those gitmate-bot warnings. Also, it has a section on commits. You currently have 4, you just need 1. :)

**Comment by Madhukar-SH** [comment]
@RohanVB I did ran coala, but I could not find any errors on local machine :/
And yah commits, I have to squash
Ty :)

**Comment by Madhukar-SH** [comment]
@meetmangukiya Thank you for the help :) Kudos

**Comment by meetmangukiya** [comment]
@Madhukar-SH squash the commits

**Comment by Madhukar-SH** [comment]
Sorry, had trouble while rebasing. All fine now :)

**Comment by RohanVB** [comment]
Hey I think @sagark123 is already assigned to the issue.

**Comment by Madhukar-SH** [comment]
@RohanVB There are 70+ files to be changed. So can't be assigned to a single person :)

**Comment by supergr35** [comment]
I feel like a better commit message would change `Usage of assertRaise was replaced by assertRaiseRegex` into `Replace the usage of assertRaise with assertRaiseRegex` followed by the rest of the commit message. This way the commit message is put in the imperative mood. 

If anyone disagrees, please notify me.

**Comment by Madhukar-SH** [comment]
@supergr35 Sure :)

**Comment by supergr35** [comment]
One small typo in the commit shortlog: `assertRaiseRegex` -> `assertRaisesRegex`
After that rebase and we should be set! 👍

**Comment by supergr35** [comment]
Weird, currently the commit shortlog (or at least the part in bold and in a bigger font above the commit body) currently says `LinterTest.py: Use assertRaiseRegex`. This should be `LinterTest.py: Use assertRaisesRegex` as that's the actual change. It appears in the message if I hover over over your commit as well.

File: tests/bearlib/abstractions/LinterTest.py [hunk 1]
```diff
@@ -154,7 +182,7 @@ def test_decorator_invalid_states(self):
 
     def test_decorator_generated_default_interface(self):
         uut = linter('some-executable')(self.ManualProcessingTestLinter)
-        with self.assertRaisesRegex(NotImplementedError, ''):
+        with self.assertRaises(NotImplementedError):
```
Comment (satwikkansal) [inline tests/bearlib/abstractions/LinterTest.py]: Unrelated change?

File: tests/bearlib/abstractions/LinterTest.py [hunk 2]
```diff
@@ -180,7 +211,10 @@ def test_decorator_invalid_parameter_types(self):
                          "The value 'invalid' for key 'critical' inside given "
                          'severity-map is no valid severity value.')
 
-        with self.assertRaises(TypeError) as cm:
+        with self.assertRaisesRegex(TypeError,
+                                    'Invalid severity value 389274234 for key '
+                                    "'critical-error' inside given severity-"
```
Comment (satwikkansal) [inline tests/bearlib/abstractions/LinterTest.py]: Just for consistency, you can use single quote and escape with "\\" wherever required :)

File: tests/bearlib/abstractions/LinterTest.py (hunk 1 above)
Comment (madhukar01) [inline tests/bearlib/abstractions/LinterTest.py]: If there is no regex to be printed, using assertRaisesRegex is of no use

Comment (satwikkansal) [inline tests/bearlib/abstractions/LinterTest.py]: Ah, better to either describe this in your commit message or make a separate issue/commit for this change :)

**Comment by userzimmermann** [comment]
Thanks @madhukar01! This is fine now :) But please also take a look at https://github.com/coala/coala/pull/3894#issuecomment-288158665 . Isn't `pytest` style looking much clearer? I would appreciate your opinion there :D

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/3949/commits/54370bcbaf4a0caba7a134f73c0392a9d74f4962

**Comment by userzimmermann** [comment]
@rultor merge

**Comment by userzimmermann** [comment]
@madhukar01 :/ Please rebase :)

**Comment by madhukar01** [comment]
@userzimmermann  Done rebasing :) Ty !

**Comment by userzimmermann** [comment]
ack https://github.com/coala/coala/pull/3949/commits/2a7d2378ad85532a2f1af40dab7c0d13821946e5

**Comment by userzimmermann** [comment]
@rultor merge
